src/ansys/geometry/core/tools/problem_areas.py

```diff
@@ -0,0 +1,395 @@
+        Server-defined ID for the body.
+    grpc_client : GrpcClient
+        Active supporting geometry service instance for design modeling.
+    edges : List[str]
+        A list of edges associated with the design.
+    """
+
+    def __init__(self, id: str, edges: List[str], grpc_client: GrpcClient):
+        """
+        Initialize a new instance of the missing face problem area class.
+
+        :param id: A unique identifier for the design.
+        :type id: str
+        :param edges: A list of edges associated with the design.
+        :type edges: List[str]
+        """
+        super().__init__(id, grpc_client)
+        self._edges = edges
+
+    @property
+    def edges(self) -> List[str]:
+        """The list of the ids of the edges connected to this problem area."""
+        return self._edges
+
+    def fix(self) -> RepairToolMessage:
+        """
+        Fix the problem area.
+
+        Returns
+        -------
+        message: RepairToolMessage
+            a message containing created and/or modified bodies.
+        """
+        id_value = Int32Value(value=int(self._id))
+        response = self._repair_stub.FixMissingFaces(
+            FixMissingFacesRequest(missing_face_problem_area_id=id_value)
+        )
+        message = RepairToolMessage(
+            response.result.success,
+            response.result.created_bodies_monikers,
+            response.result.modified_bodies_monikers,
+        )
+        return message
+
+
+class InexactEdgeProblemAreas(ProblemArea):
+    """
+    Represents an inexact edge problem area with unique identifier and associated edges.
+
+    Parameters
+    ----------
+    id : str
+        Server-defined ID for the body.
+    grpc_client : GrpcClient
+        Active supporting geometry service instance for design modeling.
+    edges : List[str]
+        A list of edges associated with the design.
+    """
+
+    def __init__(self, id: str, edges: List[str], grpc_client: GrpcClient):
+        """Initialize a new instance of the inexact edge problem area class."""
+        super().__init__(id, grpc_client)
+        self._edges = edges
+
+    @property
+    def edges(self) -> List[str]:
+        """The list of the ids of the edges connected to this problem area."""
+        return self._edges
+
+    def fix(self) -> RepairToolMessage:
+        """
+        Fix the problem area.
+
+        Returns
+        -------
+        message: RepairToolMessage
+            a message containing created and/or modified bodies.
+        """
+        id_value = Int32Value(value=int(self._id))
+        response = self._repair_stub.FixInexactEdges(
+            FixInexactEdgesRequest(inexact_edge_problem_area_id=id_value)
+        )
+        message = RepairToolMessage(
+            response.result.success,
+            response.result.created_bodies_monikers,
+            response.result.modified_bodies_monikers,
+        )
+        return message
+
+
+class ExtraEdgeProblemAreas(ProblemArea):
```

Comment:
Seems to me like this problem has not been tested, please do so.

Reply:
test_find_extra_edges,
test_find_extra_edge_id,
test_find_extra_edge_edges,
@RobPasMue these tests are for the extra edge problem areas, am I missing something?